docs/architecture.md

```diff
@@ -0,0 +1,230 @@
+## Services
+- Server:
+  - Workflow:
+    - listens for requests (source and metadata files, addresses, chain IDs)
+    - fetches potentially missing source files
+    - recompiles what users upload
+    - compares the recompiled with on-chain bytecode
+    - stores files if the comparison results in a match
+    - the match can be a `full` or a `partial` match. [Read here about the difference.](https://blog.soliditylang.org/2020/06/25/sourcify-faq#what-are-full-matches)
+  - API described in [the README](https://github.com/ethereum/sourcify#api).
+  - There is an additional, session-based API v2, designed to be used by the version of UI (UI-draft).
+  - Internal logic described in [Code Logic](#code-logic).
+- Monitor:
+  - Workflow ([link](https://medium.com/shard-labs/how-smart-contracts-can-be-automatically-verified-28ee1c5cf941)):
+    - RPC nodes are periodically queried for new blocks
+    - when a new contract is deployed, its metadata's IPFS hash is extracted from the end of the bytecode
+    - the metadata is fetched from IPFS
+    - source files information and compiler version are read from the metadata file
+    - the source files are fetched and recompiled
+    - files are stored to the repo
+- UI:
+  - provides a user friendly interface for communicating with the server
+  - two purposes:
+    - check if an address is verified
+    - verify a contract
```

Comment:
Plus inspect the source code?

Reply:
@chriseth That's what repository is for, right? UI just provides a link to the source code in the repository

Reply:
You mean by the fact that a link to the source code in the repository is generated if the checking says that the contract is indeed verified? That can be added.

Reply:
Ah right!